dg-core/src/main/java/org/finra/datagenerator/engine/scxml/SCXMLFrontier.java

```diff
@@ -80,14 +89,109 @@ public SCXMLFrontier(final PossibleState possibleState, final SCXML model) {
      * Performs a DFS on the model, starting from root, placing results in the queue
      * Just a public wrapper for private dfs function
      *
-     * @param queue the results queue
+     * @param singleThreadedProcessing Processing Strategy
+     * @param flag used to stop the search before completion
+     * @param outTemplate output Template
+     * @param sb StringBuilder with Data
+     * @return StringBuilder
+     * @throws IOException io exception
+     */
+    public StringBuilder searchForScenarios(SingleThreadedProcessing singleThreadedProcessing,
```

Comment:
It defeats the purpose of having a processing strategy interface, if one then mandates a specific implementation. This needs to work with both your single threaded processor, but also a processor correspondence to the original behavior.